fourm/pseudolabel_frames.py

```diff
@@ -2,46 +2,88 @@
+args = parser.parse_args()
+
+SHARDS = args.shards
+NTH_FRAME = args.nth_frame
+MAX_FRAMES = args.max_frames
+OUTPUT_DIR = "extracted_frames"
+LABELED_OUTPUT_DIR = "labeled_frames"
+JSON_OUTPUT_DIR = "json_output"
 
 # Ensure output directories exist
 os.makedirs(OUTPUT_DIR, exist_ok=True)
 os.makedirs(LABELED_OUTPUT_DIR, exist_ok=True)
+os.makedirs(JSON_OUTPUT_DIR, exist_ok=True)
 
 # Load the YOLO model
-model = YOLO('bbox-yolo/yolov8n.pt')  # pretrained YOLOv8n model
+model = YOLO('/cluster/work/cotterell/yemara/ml-4m/bbox-yolo/yolov8n.pt') # pretrained YOLOv8n model
 
 # Extract the tar file
 with tarfile.open(SHARDS, "r") as tar:
-    tar.extractall(path="bbox-yolo/extracted_files")
+    tar.extractall(path="extracted_files")
 
 # Iterate through extracted files
-for root, dirs, files in os.walk("bbox-yolo/extracted_files"):
+for root, dirs, files in os.walk("extracted_files"):
     for file in files:
         if file.endswith(".mp4"):
             video_path = os.path.join(root, file)
             video = cv2.VideoCapture(video_path)
-            
             frame_paths = []
-            for frame_count in range(3):  # Only process the first 3 frames to test on euler
+            frame_count = 0
+            processed_frames = 0
+
+            while True:
                 success, frame = video.read()
                 if not success:
                     break
                 
-                # Save the frame as an image
-                frame_path = os.path.join(OUTPUT_DIR, f"{file[:-4]}_frame_{frame_count}.jpg")
-                cv2.imwrite(frame_path, frame)
-                frame_paths.append(frame_path)
-            
+                if frame_count % NTH_FRAME == 0:
+                    # Save the frame as an image
+                    frame_path = os.path.join(OUTPUT_DIR, f"{file[:-4]}_frame_{frame_count}.jpg")
+                    cv2.imwrite(frame_path, frame)
+                    frame_paths.append(frame_path)
+                    processed_frames += 1
+                
+                frame_count += 1
+
+                if MAX_FRAMES and processed_frames >= MAX_FRAMES:
+                    break
+
             video.release()
             
             # Apply pseudolabeling to the extracted frames
             results = model(frame_paths, project=LABELED_OUTPUT_DIR, name=file[:-4])
             
             for i, result in enumerate(results):
+                # Save labeled image
                 result.save(filename=f'{file[:-4]}_labeled_frame_{i}.jpg')
+                
+                # Extract bounding box information
+                boxes = result.boxes
+                json_data = []
+                for box in boxes:
+                    xyxy = box.xyxy[0].tolist()  # get box coordinates
+                    conf = box.conf.item()  # get confidence score
+                    cls = int(box.cls.item())  # get class id
+                    json_data.append({
+                        "bbox": xyxy,
```

Comment:
Can we use key-names consistent with the og 4M's bounding box json key names? (I'm 90% sure it's the ones in the example here - https://github.com/swiss-ai/ml-4m/issues/11#issue-2391075201)

So for one frame that'd be like:
```
{
            "num_instances": 5,
            "image_height": 512,
            "image_width": 906,
            "instances": [
                {
                    "boxes": [
                        0.4229210317134857,
                        0.00020096010121051222,
                        0.5715101361274719,
                        0.13699540495872498
                    ],
                    "score": 0.9029952883720398,
                    "class_id": 74,
                    "class_name": "clock",
                    "segmentation": [
                        [
                            0.5055187637969095,
                            0.1337890625,
                            ...
                        ]
                    ]
                },
                {
                    "boxes": [
                        ...
                    ],
                    ...
                },
                    ...
            ]
        },
```